metaverse_modules/heal-effect/index.js

```diff
@@ -0,0 +1,426 @@
+import * as THREE from 'three';
+
+import metaversefile from 'metaversefile';
+const {useApp, useFrame, useLocalPlayer, useInternals} = metaversefile;
+const baseUrl = import.meta.url.replace(/(\/)[^\/\\]*$/, '$1');
+
+const textureLoader = new THREE.TextureLoader();
+const sparkle = textureLoader.load(`${baseUrl}/textures/sparkle.png`);
+const circle = textureLoader.load(`${baseUrl}/textures/Circle18.png`);
+const splashTexture12 = textureLoader.load(`${baseUrl}/textures/splash12.png`);
+
+export default () => {
+  const app = useApp();
+  const localPlayer = useLocalPlayer();
+  const {camera} = useInternals();
+  {
+      const flashParticleCount = 3;
+      const pixelParticleCount = 15;
+      //##################################################### get geometry #####################################################
+      const _getFlashGeometry = geometry => {
+          const geometry2 = new THREE.BufferGeometry();
+          ['position', 'normal', 'uv'].forEach(k => {
+            geometry2.setAttribute(k, geometry.attributes[k]);
+          });
+          geometry2.setIndex(geometry.index);
+          
+          const positions = new Float32Array(flashParticleCount * 3);
+          const positionsAttribute = new THREE.InstancedBufferAttribute(positions, 3);
+          geometry2.setAttribute('positions', positionsAttribute);
+          
+          const swAttribute = new THREE.InstancedBufferAttribute(new Float32Array(flashParticleCount), 1);
+          swAttribute.setUsage(THREE.DynamicDrawUsage);
+          geometry2.setAttribute('sw', swAttribute);
+
+          const scales = new Float32Array(flashParticleCount);
+          const scaleAttribute = new THREE.InstancedBufferAttribute(scales, 1);
+          geometry2.setAttribute('scales', scaleAttribute);
+
+          const id = new Float32Array(flashParticleCount);
+          const idAttribute = new THREE.InstancedBufferAttribute(id, 1);
+          geometry2.setAttribute('id', idAttribute);
+      
+          return geometry2;
+      };
+      const pixelGeometry = new THREE.BufferGeometry()
+      let group = new THREE.Group();
+      {
+          const positions = new Float32Array(pixelParticleCount * 3);
+          for(let i = 0; i < pixelParticleCount * 3; i++)
+            positions[i] = 0;
+          pixelGeometry.setAttribute('position', new THREE.BufferAttribute(positions, 3));
+
+          const opacity = new Float32Array(pixelParticleCount * 1);
+          pixelGeometry.setAttribute('opacity', new THREE.BufferAttribute(opacity, 1));
+
+          const scales = new Float32Array(pixelParticleCount * 1);
+          pixelGeometry.setAttribute('scales', new THREE.BufferAttribute(scales, 1));
+
+      }
+      //##################################################### flash material #####################################################
+      const flashMaterial = new THREE.ShaderMaterial({
```

Comment:
If you use `WebaverseShaderMaterial` (`useMaterials()`), then you don't need the log depth fragment chunks at all.

Reply:
change material to WebaverseShaderMaterial
(https://github.com/webaverse/app/pull/2873/commits/951a6e6ee293a4dd5fd5424b3ba2f599fe07a3ec)https://github.com/webaverse/app/pull/2873/commits/951a6e6ee293a4dd5fd5424b3ba2f599fe07a3ec